File: java_agent/cli/src/main/java/com/agent1/javaagent/cli/productivity/ProductivityToolCapabilities.java
package com.agent1.javaagent.cli.productivity;

import java.util.ArrayList;
import java.util.List;

/** 当前 CLI 装配的 Agent 工具能力（与 Android {@code ProductivityToolCapabilities} 文案对齐）。 */
public final class ProductivityToolCapabilities {

    private ProductivityToolCapabilities() {
    }

    public static boolean weizhiToolsOnClasspath() {
        try {
            Class.forName("com.weizhi.agent.tool.builtin.GrepTool");
            return true;
        } catch (ClassNotFoundException e) {
            return false;
        }
    }

    public static String summaryForCli(boolean weizhiScriptEnabled) {
        if (!weizhiToolsOnClasspath()) {
            return "工作区：read_file · write_file · edit_file · list_dir · chat_history（未编入 Weizhi，无脚本 / MCP）";
        }
        if (weizhiScriptEnabled) {
            return "工作区读写 + chat_history + execute_script（Weizhi）；扩展：grep / glob / zip / bash / load_skill / MCP（桌面无 WebView）";
        }
        return "工作区读写 + chat_history；扩展：grep / glob / zip / bash / load_skill / MCP（Weizhi 脚本未启用）";
    }

    public static List<String> detailBullets(boolean weizhiScriptEnabled) {
        List<String> lines = new ArrayList<>();
        lines.add("read_file · write_file · edit_file · list_dir · chat_history");
        if (!weizhiToolsOnClasspath()) {
            lines.add("未检测到 weizhi 源码 — 构建需 ./weizhi 或 AGENT1_WEIZHI_REPO");
            return lines;
        }
        if (weizhiScriptEnabled) {
            lines.add("execute_script（Weizhi 脚本，$tools 桥接）");
        } else {
            lines.add("execute_script：未启用（构建 ../weizhi native 或设置 AGENT1_WEIZHI_REPO）");
        }
        lines.add("grep · glob · zip · bash · load_skill");
        lines.add("MCP：mcp_call_tool / mcp_list_servers（配置见 .agent1/mcp_servers.json）");
        lines.add("WebView：桌面 CLI 不提供（仅 Android APK）");
        return lines;
    }
}

File: android_agent/app/src/main/java/com/dynamicui/demo/productivity/logic/business/ProductivityToolCapabilities.kt
package com.dynamicui.demo.productivity.logic.business

import com.dynamicui.demo.BuildConfig

/** 当前 APK 装配的 Agent 工具能力（与 {@link ProductivityHostAssembly} / Weizhi 子工程一致）。 */
object ProductivityToolCapabilities {

    val weizhiIntegrated: Boolean
        get() = BuildConfig.WEIZHI_INTEGRATED

    /** 设置页 / 聊天详情里展示的一行摘要。 */
    fun summaryForUi(): String = if (weizhiIntegrated) {
        "工作区读写 + chat_history + execute_script（Weizhi）；扩展：grep / glob / zip / bash / load_skill / WebView / MCP"
    } else {
        "工作区：read_file · write_file · edit_file · list_dir · chat_history（本包未编入 Weizhi，无脚本 / WebView / MCP）"
    }

    fun detailBullets(): List<String> = buildList {
        add("read_file · write_file · edit_file · list_dir · chat_history")
        if (weizhiIntegrated) {
            add("execute_script（Weizhi 脚本，\$tools 桥接）")
            add("grep · glob · zip · bash · load_skill")
            add("WebView 工具 · MCP 扩展")
        } else {
            add("未检测到 weizhi 源码或 weizhi-prebuilt/maven — WEIZHI_INTEGRATED=false")
        }
    }
}
